utils/raft.go

```
@@ -57,181 +76,1347 @@ type VoteResponse struct {
+	// Try to detect if there's already an active leader
+	foundLeader := false
+	leaderTerm := 0
+	foundLeaderID := ""
+
+	for _, peer := range rn.peers {
+		wg.Add(1)
+		go func(peer string) {
+			defer wg.Done()
+
+			// Try a quick connection to the peer
+			conn, err := grpc.Dial(peer, grpc.WithInsecure(), grpc.WithTimeout(500*time.Millisecond))
```

golangci-lint (1.64.8)

SA1019: grpc.Dial is deprecated: use NewClient instead.  Will be supported throughout 1.x. 

(staticcheck)